modules/fs2/src/main/scala/com/snowplowanalytics/snowplow/enrich/fs2/Output.scala

```diff
@@ -36,18 +36,20 @@ object Output {
   /** An item destined for the Pii Sink */
   final case class Pii[P](event: P) extends Output[Nothing, Nothing, P]
 
-  type Parsed = Output[BadRow, EnrichedEvent, EnrichedEvent]
-  type Serialized = Output[Array[Byte], Array[Byte], Array[Byte]]
+  type Attributed[B, G, P] = Output[B, AttributedData[G], AttributedData[P]]
+
+  type Parsed = Attributed[BadRow, EnrichedEvent, EnrichedEvent]
+  type Serialized = Attributed[Array[Byte], Array[Byte], Array[Byte]]
 
   /** Sends a bad/good/pii item into the appropriate sink */
   def sink[F[_]: Concurrent, B, G, P](
     bad: Pipe[F, B, Unit],
     good: Pipe[F, G, Unit],
     pii: Pipe[F, P, Unit]
   ): Pipe[F, Output[B, G, P], Unit] =
-    _.observe(_.collect { case Bad(br) => br }.through(bad))
-      .observe(_.collect { case Good(event) => event }.through(good))
-      .observe(_.collect { case Pii(event) => event }.through(pii))
+    _.observeAsync(Int.MaxValue)(_.collect { case Good(event) => event }.through(good))
```

Comment:
Are you sure we need `observeAsync` here? I'm starting to suspect this might be a performance problem we're seeing.